### PR TITLE
Attempt to optimize SVG images using SVGO

### PR DESCRIPTION
**Pull request by toolness**
This is an attempt to use [SVGO](https://github.com/svg/svgo) to optimize SVGs at build time, which helps with #1030.

### CODE REVIEW

**Comment by toolness** [comment]
Some of our SVGs, like [`icon-connect.svg`](https://github.com/mozilla/teach.webmaker.org/blob/develop/img/pages/activities/svg/icon-connect.svg?short_path=aaba15b), can't be parsed by SVGO, and the library seems to be particularly bad at handling parse errors, so I added a simple substring-based heuristic to simply log and skip them for now:

```
[13:27:24] Skipping SVG w/ Adobe entities img\pages\activities\svg\icon-connect.svg (29k)
[13:27:24] Skipping SVG w/ Adobe entities img\pages\activities\svg\icon-curriculum.svg (27k)
[13:27:24] Skipping SVG w/ Adobe entities img\pages\activities\svg\icon-learn.svg (33k)
```

Those SVGs are *really* weird, by the way. They were exported by Illustrator and use all sorts of bizarre proprietary extensions, including something called [PGF](https://en.wikipedia.org/wiki/Adobe_Illustrator_Artwork), whereby it appears to be embedding the whole original .AI file in our SVGs, thereby bloating them quite a bit.

**Comment by thisandagain** [comment]
That happens when you leave "Preserve Illustrator Editing Capabilities" checked when you export. This may sound a little odd, but it might actually be good to have SVGO fail / throw as landing SVGs with all that extra metadata should probably be considered a PR blocker.

**Comment by toolness** [comment]
Thanks @thisandagain !

I definitely agree that we should have SVGO fail ultimately--I just wanted to make it suppress the errors so I could at least get an idea of how many SVGs we needed to fix. Fortunately, only three, which I might be able to just hand-edit the proprietary goop out of...

**Comment by toolness** [comment]
Hmm, all 3 offending SVG's are from the Teaching Activities page... I believe the culprit may be @sabrinang, based on https://github.com/mozilla/teach.webmaker.org/issues/550#issuecomment-91704517 !! :see_no_evil:

**Comment by toolness** [comment]
Since those SVGs are actually just existing SVGs with different stroke/fill colors, I kinda want to fix the problem [we've talked about in the past](https://github.com/mozilla/teach.webmaker.org/issues/613#issuecomment-90721735). Probably easier to just do it manually for now, though.

**Comment by toolness** [comment]
Ok @mmmavis I think this is good for review. Aside from the code, I guess you should probably look at all the places SVGs are used and make sure that they still look okay...

File: img/pages/activities/svg/icon-connect.svg [hunk 1]
```diff
@@ -1,404 +1,25 @@
...
-AcgXRoRewXAZ1IajGzBzaD2gDSMPG3fUNRYnACDhJNAOYSUBbh5UGUIbAohZpD5CCyyxG04I0lhc
-MC6CODvipmngzLxF3slMxwYw3X9qOXqsQNAmQToi6oHWIIqwd8OkoH1K4vISfvCjrlJTrSi6Yf5r
-lyBwGM9ZPVYSMJSIyMFzSPVAw8+zoFUxfAEYgmWRFuKCQRoF0pgYgctzggStwBaAfsJKHMeDtHJD
-uEj4FnHDgC4Dw2OFvADGBXJhOXuCFD5YHsUChwQk9MQ1HhfEWwQZGtAsTKVQhHYFQisiLHbQOQSe
-5wGG0BM3hHs8LhgXVVw9cVE2eH5mXoiOrdSxazbLatCjNujiqEQmE4nHzzof8qXS6fVlJfIx7vwl
-RzuDwXDSmcgjeBP9UOTxZKjI0fHn8G/0BH6ig8fjtdN65P8BB9iOrw==
-	]]>
-</i:pgf>
-</svg>
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (toolness) [inline img/pages/activities/svg/icon-connect.svg]: I basically made these SVGs by finding non-bloated versions of them in other pages and just find/replacing the colors.

File: lib/gulp-image-converter.js [hunk 2]
```diff
@@ -41,7 +42,29 @@ function convert(options, file, cb) {
   });
 }
 
+function optimizeSvg(svgo, file, cb) {
+  var data = file.contents.toString('utf8');
+  if (data.indexOf('&ns_') !== -1) {
```
Comment (mmmavis) [inline lib/gulp-image-converter.js]: hey @toolness why are we checking `&ns_`?  I barely know anything about `SVG` and I don't know what to Google for :grimacing:

Comment (toolness) [inline lib/gulp-image-converter.js]: Oh, that's just the substring heuristic I mentioned [above](https://github.com/mozilla/teach.webmaker.org/pull/1052#issuecomment-113582058)... I've added a comment explaining things in the code!

File: lib/gulp-image-converter.js [hunk 3]
```diff
@@ -41,7 +42,44 @@ function convert(options, file, cb) {
...
+  // character entities that begin with "&ns_". If we pass this on to svgo,
+  // it will be confused by the entities and throw horribly confusing
+  // errors, so instead we'll detect for the condition here and bail early
+  // with an informative help message.
+  //
+  // For more information, see:
+  //
+  // https://github.com/mozilla/teach.webmaker.org/pull/1052#issuecomment-113582058
+
+  if (data.indexOf('&ns_') !== -1) {
```
Comment (mmmavis) [inline lib/gulp-image-converter.js]: (I thought I left a comment here few days ago... not sure why it didn't go through)
@toolness I barely know anything about SVG so I wasn't able to google what "&ns_" represents.   Curious what this check is for :eyes:

**Comment by mmmavis** [comment]
@toolness ah nvm, just saw this: https://github.com/toolness/teach.webmaker.org/commit/04bff26abbbf28439ecba44c908f64aac84f05a5 :+1:

File: lib/gulp-image-converter.js (hunk 3 above)
Comment (mmmavis) [inline lib/gulp-image-converter.js]: @toolness ah nvm, just saw this: https://github.com/toolness/teach.webmaker.org/commit/04bff26abbbf28439ecba44c908f64aac84f05a5 :+1:

**Comment by mmmavis** [comment]
will get to this PR today :smiley:

**Comment by mmmavis** [comment]
Wooohooooo this is awesome @toolness !

I tried adding a SVG that was exported with "Preserve Illustrator Editing Capabilities" checked and the test suite did raise a flag

```bash
Error: SVG was exported with "Preserve Illustrator Editing Capabilities" checked: img/pages/events/test.svg
    at /Users/mavisou/Development/teach.webmaker.org/lib/gulp-image-converter.js:62:10
    at process._tickCallback (node.js:355:11)
```

:tada: :tada: :tada:

**Comment by mmmavis** [comment]
@sabrinang you might want to update the [3 SVGs](https://github.com/mozilla/teach.webmaker.org/pull/1052/files) on Google Drive as well :stuck_out_tongue_closed_eyes:

**Pull request merged**